pkg/scheduler/framework/plugins/noderesources/resource_allocation.go

```diff
@@ -35,14 +37,43 @@ type resourceAllocationScorer struct {
 	Name string
 	// used to decide whether to use Requested or NonZeroRequested for
 	// cpu and memory.
-	useRequested        bool
-	scorer              func(requested, allocable resourceToValueMap) int64
-	resourceToWeightMap resourceToWeightMap
+	useRequested           bool
+	scorer                 func(requested, allocatable resourceToValueMap) int64
+	resourceToWeightMap    resourceToWeightMap
+	resourceToValueMapPool *sync.Pool
+}
+
+func newResourceAllocationScorer(name string, scorer func(resourceToValueMap, resourceToValueMap) int64,
+	r resourceToWeightMap, useRequested bool) *resourceAllocationScorer {
+	return &resourceAllocationScorer{
+		Name:                name,
+		scorer:              scorer,
+		resourceToWeightMap: r,
+		useRequested:        useRequested,
+		resourceToValueMapPool: &sync.Pool{
+			New: func() interface{} {
+				return resourceToValueMap(make(map[v1.ResourceName]int64, 4))
```

Comment:
How did 4 come from?  Did you use other different values here for the evaluation?